### PR TITLE
ag3.snp_allele_frequencies() 

### PR DESCRIPTION
**Pull request by cclarkson**
Given a transcript, a populations of interest (dict), site_mask, site_filters, species_calls and sample_sets returns a dataframe with allele frequencies per variant per population.

Tests are included.

### CODE REVIEW

**Comment by cclarkson** [comment]
Hi @alimanfoo, this is ready for another review.

File: malariagen_data/ag3.py [hunk 1]
```diff
@@ -713,61 +726,113 @@ def snp_effects(self, transcript, site_mask):
...
+            cols[pop] = afs[pop][:, 1:].flatten()
+
+        # build df
+        df = pandas.DataFrame(cols)
+
+        # drop invariants
+        if drop_invariants:
+            df['dropping'] = df[populations].sum(axis=1)
+            df = df[df.dropping > 0]
+            df.drop('dropping', axis=1, inplace=True)
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:830]:
```suggestion
            loc_variant = df[populations].sum(axis=1) > 0
            df = df[loc_variant]
```

Simplify, there's no need to add then remove a column here.

File: malariagen_data/ag3.py [hunk 2]
```diff
@@ -713,61 +726,113 @@ def snp_effects(self, transcript, site_mask):
...
+            "bf_2012_col": "country == 'Burkina Faso' and year == 2012 and species == 'coluzzii'"
+        site_mask : {"gamb_colu_arab", "gamb_colu", "arab"}
+            Site filters mask to apply.
+        site_filters : str, optional
+            Site filters analysis version.
+        sample_sets : str or list of str, optional
+            Can be a sample set identifier (e.g., "AG1000G-AO") or a list of sample set
+            identifiers (e.g., ["AG1000G-BF-A", "AG1000G-BF-B"]) or a release identifier (e.g.,
+            "v3") or a list of release identifiers.
+        drop_invariants : bool, optional
```
Comment (alimanfoo) [inline malariagen_data/ag3.py:755]:
```suggestion
        drop_invariant : bool, optional
```

Nit, aesthetics.

File: tests/test_ag3.py [hunk 3]
```diff
@@ -576,3 +576,66 @@ def test_snp_effects():
...
+    assert expected_fields == df.columns.tolist()
+    assert df.shape == (132306, 6)
+    assert df.loc[0].position == 2358158
+    assert df.loc[1].ref_allele == "A"
+    assert df.loc[2].alt_allele == "G"
+    assert df.loc[3].gm == 0.0
+    assert df.loc[4].mz == 0.0
+    assert df.loc[72].maximum == pytest.approx(0.001792, abs=1e-6)
+    # check invariant positions are still present
+    assert 0 in df.maximum.unique()
```
Comment (alimanfoo) [inline tests/test_ag3.py:641]:
```suggestion
    assert np.any(df.maximum == 0)
```

Slightly clearer perhaps.

**Review by alimanfoo — COMMENTED** [review]
Thanks Chris, looking great, only some very small stylistic comments.

Also please could you run black on the source code.

**Comment by alimanfoo** [comment]
One more thing, there is a `print()` statement in the `Ag3.snp_effects()` method left over from debugging, let's remove that.

**Pull request merged**